packages/block-library/src/audio/edit.js

```diff
@@ -105,51 +114,67 @@ class AudioEdit extends Component {
 	render() {
 		const { autoplay, caption, loop, preload, src } = this.props.attributes;
 		const { setAttributes, isSelected, className, noticeOperations, noticeUI } = this.props;
-		const { editing } = this.state;
-		const switchToEditing = () => {
-			this.setState( { editing: true } );
+		const { isEditing } = this.state;
+		const toggleIsEditing = () => {
+			this.setState( { isEditing: ! this.state.isEditing } );
+			if ( this.state.isEditing ) {
+				speak( __( 'You can now listen to the audio in the audio block.' ) );
+			} else {
+				speak( __( 'You are now editing the audio in the audio block.' ) );
+			}
 		};
 		const onSelectAudio = ( media ) => {
 			if ( ! media || ! media.url ) {
-				// in this case there was an error and we should continue in the editing state
+				// in this case there was an error and we should continue in the isEditing state
 				// previous attributes should be removed because they may be temporary blob urls
 				setAttributes( { src: undefined, id: undefined } );
-				switchToEditing();
+				toggleIsEditing();
 				return;
 			}
 			// sets the block's attribute and updates the edit component from the
-			// selected media, then switches off the editing UI
+			// selected media, then toggles off the isEditing UI
 			setAttributes( { src: media.url, id: media.id } );
-			this.setState( { src: media.url, editing: false } );
+			this.setState( { src: media.url, isEditing: false } );
 		};
-		if ( editing ) {
-			return (
-				<MediaPlaceholder
-					icon={ <BlockIcon icon={ icon } /> }
-					className={ className }
-					onSelect={ onSelectAudio }
-					onSelectURL={ this.onSelectURL }
-					accept="audio/*"
-					allowedTypes={ ALLOWED_MEDIA_TYPES }
-					value={ this.props.attributes }
-					notices={ noticeUI }
-					onError={ noticeOperations.createErrorNotice }
+		const editImageIcon = ( <SVG width={ 20 } height={ 20 } viewBox="0 0 20 20"><Rect x={ 11 } y={ 3 } width={ 7 } height={ 5 } rx={ 1 } /><Rect x={ 2 } y={ 12 } width={ 7 } height={ 5 } rx={ 1 } /><Path d="M13,12h1a3,3,0,0,1-3,3v2a5,5,0,0,0,5-5h1L15,9Z" /><Path d="M4,8H3l2,3L7,8H6A3,3,0,0,1,9,5V3A5,5,0,0,0,4,8Z" /></SVG> );
+		const getEditToolbar = ( isActive ) => {
```

Comment:
How about making this a component that can be used as `<AudioToolbar isEditing={ isEditing } />`? The implementation would be pretty much the same, as it'd be a function component. It'd require an extra prop for `toggleIsEditing`. I would also maybe move the implementation into a separate file, or to the top of this file. `editImageIcon` would have to be moved with it.

Reply:
done, looks nicer! Thanks! :)